app/Repositories/SignupRepository.php

```diff
@@ -19,7 +19,7 @@ public function create($data, $northstarId)
         $signup = new Signup;
 
         $signup->northstar_id = $northstarId;
-        $signup->campaign_id = $data['campaign_id'];
+        $signup->campaign_id = isset($data['campaign_id']) ? $data['campaign_id'] : get_campaign_by_action_id($data['action_id'])->id;
```

Comment:
Can the `campaign_id` kind of trickle down to here once it's grabbed from the action the first time? If not, is something like this needed for posts as well?

Reply:
ah great call, updated that @katiecrane ! 

> is something like this needed for posts as well?

We don't need this because when a post is created, it [grabs the `campaign_id` from the signup it belongs to](https://github.com/DoSomething/rogue/blob/master/app/Repositories/PostRepository.php#L99)!